interview-questions.md

```diff
@@ -10,58 +10,71 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 
 3. Which lifecycle method is required in a React class component?
 
   Your answer:
+  The constructor method is required when creating data that belongs to a class. It runs automatically when a class is instantiated. For example when you're setting up a component and instantiating the class, you would type "constructor(){} within the class's code block.
 
   Researched answer:
+During the "Mounting" component lifecycle, the constructor method is required and called before anything else in the component. It's called with "props", as arguments and should always call super(props) before anything else which will initiate the parent class's constructor method and allows the component to inherit methods from the parent class.
 
 
 
 4. What is JSX? What is one notable difference between HTML and JSX?
 
   Your answer:
+  JSX is like the javascript version of html. It's a markup language that pre processes HTML (actually XML) into plain javascript before the file is sent to the browser. The render method in react uses JSX and it's encapsulated by "fragments" which are empty tags that look like this <></>. One big difference between HTML and JSX is that the JSX in React only returns one element. For example if you put a header and a paragraph tag by itself in the JSX, it will only render the first thing, the header. In order to render all of that, you will need to encapsulate all of those elements into a div.
 
   Researched answer:
+  JSX is a markup that allows the developer to write HTML into javascript and allow easy placement in the DOM. It is not required but highly recommended to use JSX as it's a helpful visual aid when modifying the UI inside javascript code. One big difference is that you would need to encapsulate all HTML elements within a div if you want all elements to be rendered properly in the JSX
 
 
 
 5. What is yarn? What file(s) are modified in a React application when you run yarn?
 
   Your answer:
+  YARN stands for Yet Another React Negotiator. It's a package manager that creates the file structure for a React app. You can setup a new React app via the terminal with yarn commands. When you run yarn, it will install the base file structure and dependencies needed for your React app. When modifying the code of the React application, we make a components folder under src where we create and code new components and import them within the App.js of the app's root folder.
 
   Researched answer:
-
+Yarn is a packaging system developed for node.js. Yarn is used to create new React applications and will build the file structure for any new app via terminal. Using yarn to run the React application is similar to running a server, and the application can be stopped by hitting ctrl + c in the terminal. yarn create react-app "name-of-app" creates the director and installs the files and dependencies needed for the app. The developer then makes a components folder under src and also modifies App.js when developing the React app.
```

Comment:
Lots of great content. Yarn definitely saves us time by bringing all the necessary code snippets to get our coding projects to properly function.